src/tools/CMakeLists.txt

```diff
@@ -25,5 +26,6 @@ binaryen_add_executable(wasm-as wasm-as.cpp)
 binaryen_add_executable(wasm-dis wasm-dis.cpp)
 binaryen_add_executable(wasm-ctor-eval wasm-ctor-eval.cpp)
 binaryen_add_executable(wasm-reduce wasm-reduce.cpp)
+binaryen_add_executable(binaryen-fuzz-types "${fuzzing_SOURCES};fuzzing/binaryen-fuzz-types.cpp")
```

Comment:
I'm unclear on why we want a new executable here? (And also why we want one with a new prefix, `binaryen-`, and that it is the first whose main file resides outside of `tools/`). In general I think we have a high bar for adding new executables, as we don't want to keep accumulating them over time.

What is the purpose of the new executable?

If the purpose is mainly to simplify testing, could we add a cpp unit test that does the same as the executable?

Reply:
It seemed better to create a new executable than to clutter wasm-opt with new functionality unrelated to anything else it does. The purpose of the new executable is to fuzz the type system specifically, and doing that separately from the current fuzzing infrastructure lets us fuzz thousands of iterations per second. I imagine that in the future we could add additional fuzzers of specific subsystems to this tool for the same benefit. I chose the `binaryen-` prefix to try to hint that this is an internal development tool that is probably not generally useful, but that could probably be clearer. I'm open to suggestions on the name. I can also move the executable main file out to the `tools` directory.

Reply:
I see, thanks.

What does LLVM do for such "mostly-internal" tools?

Reply:
LLVM doesn't have very clear conventions here. Many LLVM tools start with the `llvm-` prefix, but both "internal" tools such as `llvm-mc` and drop-in replacements for well-known tools such as `llvm-cxxfilt` follow that convention. Other LLVM tools have entirely different naming schemes, such as `obj2yaml`, which does not hint at an LLVM connection at all.

Reply:
I see. Yeah, `obj2yamp` and `opt` are weird to not have `llvm-` prefixes... anyhow, if they have "internal" tools with their prefix, then I see no reason for us not to either. I think I'd prefer `wasm-fuzz*` over starting to consider what a special prefix might look like and other bikeshedding about the location of the files etc.

Reply:
Sure, I'll rename to `wasm-fuzz-types` (and we can make it more general if we want in the future) and move the main file up to `tools`.